cmake/BLTMacros.cmake

```diff
@@ -438,12 +455,20 @@ macro(blt_add_library)
         endif()
     endforeach()
     if(_have_fortran)
-      target_include_directories(${arg_NAME} PRIVATE ${CMAKE_Fortran_MODULE_DIRECTORY})
+        target_include_directories(${arg_NAME} PRIVATE ${CMAKE_Fortran_MODULE_DIRECTORY})
     endif()
 
     blt_setup_target( NAME ${arg_NAME}
                       DEPENDS_ON ${arg_DEPENDS_ON} )
 
+    if ( arg_INCLUDES )
```

Comment:
Do we need to distinguish here between header only (``INTERFACE``) and compiled (``PUBLIC``) libraries?

Reply:
Can we use `INTERFACE` all the time?

Reply:
This needs to be more thoroughly tested. I spoke to kenny about it and we agreed to not hold this PR up due to possible problems with header-only libraries.  Switching this to INTERFACE would cause problems though as the includes would not be used for the current library.